core/network/src/service.rs

```diff
@@ -275,13 +288,14 @@ impl<B: BlockT + 'static, S: NetworkSpecialization<B>> Drop for Service<B, S> {
 }
 
 impl<B: BlockT + 'static, S: NetworkSpecialization<B>> SyncProvider<B> for Service<B, S> {
+	fn is_major_syncing(&self) -> bool {
+		self.is_major_syncing()
+	}
 	/// Get sync status
-	fn status(&self) -> ProtocolStatus<B> {
-		let (sender, port) = channel::unbounded();
-		let _ = self.protocol_sender.send(ProtocolMsg::Status(sender));
-		port.recv().expect("1. Protocol keeps handling messages until all senders are dropped,
-			or the ProtocolMsg::Stop message is received,
-			2 Service keeps a sender to protocol, and the ProtocolMsg::Stop is never sent.")
+	fn status(&self) -> mpsc::UnboundedReceiver<ProtocolStatus<B>> {
+		let (sink, stream) = mpsc::unbounded();
+		self.status_sinks.lock().push(sink);
```

Comment:
A note for future, api wise there is a risk of making the Vec grow (there is no removal on receiver drop or on send fail or any way it can be cleaned up), but with the current usage I think there is not much risk.